src/forge/portablemc_forge/__init__.py

```diff
@@ -195,19 +198,13 @@ def prepare(self):
             parent_version.prepare_jvm()
             self.jvm_exec = parent_version.jvm_exec
 
-    def download(self):
-
-        if self.main_dir is None:
-            raise ValueError()
-
-        self.check_download(self.dl.download_files())
-
     def check_download(self, report: DownloadReport):
 
         installer_fails_count = 0
+        i = 0
         for entry, entry_fail in report.fails.items():
-            if entry.dst == self.installer_file:
```

Comment:
This condition is really important, because the `dl` (`DownloadList`) is also used for the parent version, which might add its own files to download